platform/qt/src/qmapboxgl.cpp

```diff
@@ -964,15 +1152,30 @@ void QMapboxGL::updateSource(const QString &sourceID, const QVariantMap &params)
     }
 }
 
-void QMapboxGL::removeSource(const QString& sourceID)
+/*!
+    Removes the source \a id.
+
+    Has no effect if the source does not exist.
+*/
+void QMapboxGL::removeSource(const QString& id)
 {
-    auto sourceIDStdString = sourceID.toStdString();
+    auto sourceIDStdString = id.toStdString();
 
     if (d_ptr->mapObj->getSource(sourceIDStdString)) {
         d_ptr->mapObj->removeSource(sourceIDStdString);
     }
 }
 
+/*!
+    Adds a custom layer \a id with the initialization function \a initFn, the
+    render function \a renderFn and the deinitialization function \a deinitFn with
+    the user data \a context_ before the existing layer \a before.
+
+    This is used for delegate the rendering to the user of this API and is not
+    officially supported. Use at your own risk.
+
+    \deprecated
```

Comment:
I tried to find something like `\unsupported` tag in the same lines as `\deprecated` on `qdoc` but could not find. Making it as `\deprecated` at least moves it to a almost hidden section of the generated HTML.

I do agree with you, it is misleading. I'm going to add a `\warning` tag.

There is no such thing as unsupported public API on Qt.